### ISSUE TITLE
"Circular reference detected" when outputting as JSON

### ISSUE THREAD

**Issue by BenjaminHae** [nabla-c0d3/sslyze#375]
**Describe the bug**
When using the JSON output, sslyze fails with an error indicating a "circular reference".

**To Reproduce**
1. Install SSLyze using pip
2. Run the following command `sslyze --regular --json_out=test.json www.google.de:443`
4. See error after scan completed:
```
Traceback (most recent call last):
  File "/usr/local/bin/sslyze", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/sslyze/__main__.py", line 117, in main
    output_hub.scans_completed(exec_time)
  File "/usr/local/lib/python3.7/dist-packages/sslyze/cli/output_hub.py", line 69, in scans_completed
    out_generator.scans_completed(total_scan_time)
  File "/usr/local/lib/python3.7/dist-packages/sslyze/cli/json_output.py", line 79, in scans_completed
    self._json_dict, default=_object_to_json_dict, sort_keys=True, indent=4, ensure_ascii=True
  File "/usr/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.7/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/usr/lib/python3.7/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 439, in _iterencode
    yield from _iterencode(o, _current_indent_level)
  File "/usr/lib/python3.7/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 439, in _iterencode
    yield from _iterencode(o, _current_indent_level)
  File "/usr/lib/python3.7/json/encoder.py", line 436, in _iterencode
    raise ValueError("Circular reference detected")
ValueError: Circular reference detected
```
5. `test.json` exists, but is empty.
**Expected behavior**
No error, export of data as JSON.

**Python environment (please complete the following information):**
 - OS: Kali Linux Rolling
 - Python version: Python 3.7.3
 - sslyze version: 2.1.1

**Comment by fbexiga** [comment]
Having the same issue.

**Comment by nabla-c0d3** [comment]
Fixed in 2.1.2.
